src/Week4/HW04.hs

```diff
@@ -0,0 +1,103 @@
+{-# OPTIONS_GHC -Wall #-}
```

Comment:
Instead of adding this language pragma to each source file, you can just add the following line to cabal file, under the `library` section: `ghc-options: -Wall`